controllers/machinehealthcheck_controller.go

```diff
@@ -367,10 +348,17 @@ func (r *MachineHealthCheckReconciler) getMachineFromNode(nodeName string) (*clu
 	); err != nil {
 		return nil, errors.Wrap(err, "failed getting machine list")
 	}
-	if len(machineList.Items) != 1 {
-		return nil, errors.Errorf("expecting one machine for node %v, got: %v", nodeName, machineList.Items)
+	items := []*clusterv1.Machine{}
+	for i := range machineList.Items {
+		machine := &machineList.Items[i]
+		if machine.Status.NodeRef != nil && machine.Status.NodeRef.Name == nodeName {
+			items = append(items, machine)
+		}
+	}
```

Comment:
Does this not defeat the point of the `machineNodeNameIndex` included in the list call above? I feel like we are having to add code here to make the tests happy because fake client doesn't support indexing

Reply:
I can revert this, the tests pass regardless of this loop or not which makes me think we're missing some tests, although I don't have enough time today to dig into it more.

I'll add a TODO on top of this to remove this loop once the fake client supports indexing, how does that sound?

Reply:
Yeah that sounds good, add a todo, I might see if I can work out why they pass without this, seems odd 🤔

Reply:
Done 👍, thanks for looking into it